legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/depot/model/DepotProjectId.java

```diff
@@ -0,0 +1,101 @@
+// Copyright 2021 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.sdlc.server.depot.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class DepotProjectId
+{
+    private static final char DELIMITER = ':';
+
+    private final String groupId;
+    private final String artifactId;
+
+    private DepotProjectId(String groupId, String artifactId)
+    {
+        this.groupId = groupId;
+        this.artifactId = artifactId;
+    }
+
+    public String getGroupId()
+    {
+        return this.groupId;
+    }
+
+    public String getArtifactId()
+    {
+        return this.artifactId;
+    }
+
+    @Override
+    public boolean equals(Object other)
+    {
+        if (this == other)
+        {
+            return true;
+        }
+        if (!(other instanceof DepotProjectId))
+        {
+            return false;
+        }
+        DepotProjectId that = (DepotProjectId)other;
+        return this.getGroupId().equals(that.getGroupId()) && this.getArtifactId().equals(that.getArtifactId());
+
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return this.groupId.hashCode() ^ this.artifactId.hashCode();
+    }
+
+    @Override
+    public String toString()
+    {
+        return String.format("%s%s%s", this.groupId, DELIMITER, this.artifactId);
```

Comment:
You may want to consider how you'll handle null values here. Also, string concatenation is simpler and clearer for this case.